MSAL/src/MSALPublicClientApplication.m

```diff
@@ -196,123 +211,106 @@ - (id)initWithClientId:(NSString *)clientId
-        return nil;
-    }
-
-    MSALRedirectUri *msalRedirectUri = [MSALRedirectUriVerifier msalRedirectUriWithCustomUri:redirectUri
-                                                                                    clientId:clientId
+    MSALRedirectUri *msalRedirectUri = [MSALRedirectUriVerifier msalRedirectUriWithCustomUri:config.redirectUri
+                                                                                    clientId:config.clientId
                                                                                        error:&msidError];
-
+    
     if (!msalRedirectUri)
     {
         if (error) *error = [MSALErrorConverter msalErrorFromMsidError:msidError];
         return nil;
     }
-
-    _redirectUri = msalRedirectUri;
-
+    
+    config.verifiedRedirectUri = msalRedirectUri;
+    
 #if TARGET_OS_IPHONE
     // Optional Paramater
-    _keychainGroup = keychainGroup;
-
-    MSIDKeychainTokenCache *dataSource;
-    if (_keychainGroup == nil)
-    {
-        _keychainGroup = [[NSBundle mainBundle] bundleIdentifier];
-    }
+    MSIDKeychainTokenCache *dataSource = [[MSIDKeychainTokenCache alloc] initWithGroup:config.cacheConfig.keychainSharingGroup];
     
-    dataSource = [[MSIDKeychainTokenCache alloc] initWithGroup:_keychainGroup];
-
     MSIDLegacyTokenCacheAccessor *legacyAccessor = [[MSIDLegacyTokenCacheAccessor alloc] initWithDataSource:dataSource otherCacheAccessors:nil];
     MSIDDefaultTokenCacheAccessor *defaultAccessor = [[MSIDDefaultTokenCacheAccessor alloc] initWithDataSource:dataSource otherCacheAccessors:@[legacyAccessor]];
     
     self.tokenCache = defaultAccessor;
-    
-    _webviewType = MSALWebviewTypeDefault;
-    
 #else
     __auto_type dataSource = MSIDMacTokenCache.defaultCache;
-
+    
     MSIDDefaultTokenCacheAccessor *defaultAccessor = [[MSIDDefaultTokenCacheAccessor alloc] initWithDataSource:dataSource otherCacheAccessors:nil];
     self.tokenCache = defaultAccessor;
-    _webviewType = MSALWebviewTypeWKWebView;
-    
 #endif
-    
-    _validateAuthority = YES;
-    _extendedLifetimeEnabled = NO;
-    
-    _sliceParameters = [MSALPublicClientApplication defaultSliceParameters];
+    _configuration = config;
     
     MSIDAADNetworkConfiguration.defaultConfiguration.aadApiVersion = @"v2.0";
-    _expirationBuffer = 300;  //in seconds, ensures catching of clock differences between the server and the device
     
     return self;
 }
 
+- (id)initWithClientId:(NSString *)clientId
+         keychainGroup:(NSString *)keychainGroup
+             authority:(MSALAuthority *)authority
+           redirectUri:(NSString *)redirectUri
+                 error:(NSError * __autoreleasing *)error
+{
+    if (!(self = [super init]))
+    {
+        return nil;
+    }
+    MSALPublicClientApplicationConfig *config = [[MSALPublicClientApplicationConfig alloc] initWithClientId:clientId redirectUri:redirectUri authority:authority];
+    
+#if TARGET_OS_IPHONE
```

Comment:
nit: wasn't this overload only available on iPhone already?

Reply:
CORRECT!